src/Nat80.mo

```diff
@@ -34,7 +34,7 @@ module {
 		var l = n80.low +% n;
 		var h = n80.high;
 		if (n80.low < l) h +%= 1;
-		if (n80.high < h) return #err("overflow");
+		if (n80.high > h) return #err("overflow");
```

Comment:
It seems like both brackets need to be inverted.
```
if (l < n80.low) h +%= 1;
if (h < n80.high) return #err("overflow");
```

Reply:
Ah good point, I will test that out just to make sure and then push up a fix